### PR TITLE
The actual gutlunch fix

### PR DESCRIPTION
**Pull request by TerraGS**
## About The Pull Request
Adds an override for ListTargetsLazy so gutlunch will check for wanted objects to toggle AI on and not just mobs.

## Why It's Good For The Game
Should finally do what is intended since I understand what is going on this time. Also fixed my prior mistake on the first CanAttack condition, which is meant to prevent the mob from attacking a floor.

### CODE REVIEW

**Comment by TerraGS** [comment]
I must restore my honor

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm [hunk 1]
```diff
@@ -81,6 +81,14 @@
 		add_overlay(gutlunch_full_overlay)
 	..()
 
+/mob/living/simple_animal/hostile/asteroid/gutlunch/ListTargetsLazy(_Z)//override to include wanted_objects as valid targets
+	. = ..()
+	for (var/I in view(vision_range, loc))
+		var/obj/O = I
```
Comment (Kapu1178) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm:87]:
```suggestion
	for (var/atom/thing as anything  in view(vision_range, loc))
```

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm [hunk 2]
```diff
@@ -81,6 +81,14 @@
 		add_overlay(gutlunch_full_overlay)
 	..()
 
+/mob/living/simple_animal/hostile/asteroid/gutlunch/ListTargetsLazy(_Z)//override to include wanted_objects as valid targets
+	. = ..()
+	for (var/I in view(vision_range, loc))
```
Comment (tralezab) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm:86]: for `var/I` in view will grab EVERYTHING in view. That means non objects...

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm (hunk 1 above)
Comment (tralezab) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm:87]: which are typecasted as objects here.

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm (hunk 2 above)
Comment (tralezab) [reply code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm:86]: Also, `I` is a nondescript var name.

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm (hunk 1 above)
Comment (tralezab) [reply code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm:87]: Also, `O` is a nondescript var name.

**Review by tralezab — APPROVED** [review]
Well, it does fix the issue. but obviously, why don't we just not write more simplemob ai shitcode in the first place? This is a bandaid, as I see it

**Comment by tralezab** [comment]
undraft when ready

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm (hunk 1 above)
Comment (Kapu1178) [reply code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm:87]: `view(x, turf/obj)` will return movables only

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm [hunk 3]
```diff
@@ -81,6 +81,14 @@
...
+		var/obj/O = I
+		if (is_type_in_typecache(O, wanted_objects))
```
[the remark this answers is not in the document]
Comment (Kapu1178) [reply code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm:88]: this is faster

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm [hunk 4]
```diff
@@ -81,6 +81,14 @@
...
+			if (isturf(O.loc))
+				. += O
```
Comment (Kapu1178) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm:90]: i *think* this is unnecessary if my changes are applied, not 100% sure though.

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm [hunk 5]
```diff
@@ -81,6 +81,14 @@
 		add_overlay(gutlunch_full_overlay)
 	..()
 
+/mob/living/simple_animal/hostile/asteroid/gutlunch/ListTargetsLazy(_Z)//override to include wanted_objects as valid targets
```
[the remark this answers is not in the document]
Comment (Kapu1178) [reply code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm:84]: confuses with src.z

**Comment by itseasytosee** [comment]
Too much heat?

**Pull request closed without merging**